itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestStatsReplicationScenarios.java

```diff
@@ -359,17 +383,20 @@ private void testStatsReplicationCommon(boolean parallelBootstrap, boolean metad
   }
 
   @Test
-  public void testForNonAcidTables() throws Throwable {
+  public void testNonParallelBootstrapLoad() throws Throwable {
+    LOG.info("Testing " + testName.getClass().getName() + "." + testName.getMethodName());
     testStatsReplicationCommon(false, false);
   }
 
   @Test
-  public void testForNonAcidTablesParallelBootstrapLoad() throws Throwable {
-    testStatsReplicationCommon(true, false);
+  public void testForParallelBootstrapLoad() throws Throwable {
+    LOG.info("Testing " + testName.getClass().getName() + "." + testName.getMethodName());
+    testStatsReplicationCommon(true, false );
   }
 
   @Test
-  public void testNonAcidMetadataOnlyDump() throws Throwable {
+  public void testMetadataOnlyDump() throws Throwable {
```

Comment:
Add more tests for the following scenarios.
1. REPL LOAD fails after replicating table or partition objects with stats but before setting last replId. Now, retry which takes alter table/partition replace flows and stats should be valid after successful replication. Need this for all non-transactional, transactional and migration cases.
2. Parallel inserts with autogather enabled. Now, we will have events such that multiple txns open when updating stats event. Also, try to simulate that one stats update was successful and the other one invalidates it due to concurrent writes.

Reply:
Added test for the first case. For second case, the events for parallel inserts will be serialized and applied serially on repl side. So this should be a problem on repl. We may test whether the events are generated in serialized fashion and have same expected contents. But that should be done a test which tests concurrent inserts (may be we already have it somewhere) and not in a replication test.